front/src/modules/timesStops/helpers/computeMargins.ts

```diff
@@ -30,47 +26,55 @@ function getTheoreticalMargin(selectedTrainSchedule: TrainScheduleResult, pathSt
   return selectedTrainSchedule.margins!.values[theoreticalMarginBoundaryIndex + 1];
 }
 
-function computeDuration(
-  trainSimulation: ReportTrain,
-  opPoint: OperationalPointWithTimeAndSpeed,
-  nextOpPoint: OperationalPointWithTimeAndSpeed
-) {
-  const opPointTime = interpolateValue(trainSimulation, mToMm(opPoint.position), 'times');
-  const nextOpPointTime = interpolateValue(trainSimulation, mToMm(nextOpPoint.position), 'times');
-  return msToS(nextOpPointTime - opPointTime);
-}
-
 function computeMargins(
-  trainSimulation: SimulationResponseSuccess,
-  opPoint: OperationalPointWithTimeAndSpeed,
-  nextOpPoint: OperationalPointWithTimeAndSpeed,
   selectedTrainSchedule: TrainScheduleResult,
-  pathStepId: string
+  pathStepIndex: number,
+  pathItemTimes: NonNullable<TrainScheduleWithDetails['pathItemTimes']> // in ms
 ) {
+  const { path, margins } = selectedTrainSchedule;
+  if (
+    !margins ||
+    (margins.values.length === 1 && margins.values[0] === '0%') ||
+    pathStepIndex === selectedTrainSchedule.path.length - 1
+  ) {
+    return {
+      theoreticalMargin: undefined,
+      theoreticalMarginSeconds: undefined,
+      calculatedMargin: undefined,
+      diffMargins: undefined,
+    };
+  }
+
+  const pathStepId = path[pathStepIndex].id;
   const theoreticalMargin = getTheoreticalMargin(selectedTrainSchedule, pathStepId);
 
-  // durations to go from current scheduled point to next scheduled point.
+  // find the previous pathStep where margin was defined
+  let prevIndex = 0;
+  // eslint-disable-next-line no-plusplus
+  for (let index = 1; index < pathStepIndex; index++) {
```

Comment:
Why disable the eslint warning instead of doing index+=1?

```suggestion
  for (let index = 1; index < pathStepIndex; index += 1) {
```

Reply:
You're right, thanks :)